### PR TITLE
Create snap-to-grid and translation fixes

### PR DESCRIPTION
**Pull request by ctrlaltdavid**
Create app fixes:
- Entity y value jumping when translate with snap-to-grid enabled.
- Oscillating positons when translating with snap-to-grid enabled.
- Translation handle not following mouse at some view angles.

### CODE REVIEW

**Comment by ctrlaltdavid** [comment]
### Test Plan

In a domain where you have edit permissions.
Run the Create app.
To toggle between local and world axes for translation axes, press Ctrl-T. This is only possible if a single entity is selected; multiple entities always use world axes.

Enable snap-to-grid: "GRID" tab of the Create dialog > "Snap entities to grid". When an entity is translated using the translation handles, its coordinate values should jump in multiples of the minor grid size (e.g., if x = 10.3 and the minor unit is 1, the snap positions should be ..., 9.3, 10.3, 11.3 ...).

#### A: Desktop mode.

1) Check that a a single entity's y position value does not jump when start translating on any x or z axes, x-z plane (click-hold-move entity rather than handle).
a) Test local axes mode; x and z translation handles plus x-z plane.
b) Test world axes mode; x and z translation handles plus x-z plane.

2) Check that the a single entity's postion values change in multiples of the minor grid size.
a) Test local axes mode; x, y, and z translation handles; plus x-z plane.
b) Test world axes mode; x, y, and z translation handles; plus x-z plane.

3) Check that mulptiple entities' position values change in multiples of the minor grid size. (World axes mode only.)
X, y, and z translation handles; plus x-z plane.

Disable snap-to-grid.

4) Check that translation handle follows mouse cursor when camera is at an angle.
- Reproduce the problem with recent master: move a distance away and up above the entity; right-click-drag on display to tilt camera down to look at entity; test each translate handle: for at least one, the entity should move haphazardly with the translation handle not following the mouse.
- Check that the problem is fixed in the PR build.

5) Check X-Z translation works OK.

#### B: HMD mode.

Repeat 1) .. 5) in HMD mode.

https://highfidelity.manuscript.com/f/cases/16392

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11123/HighFidelity-Beta-PR13543-a642324.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11123/HighFidelity-Beta-PR13543-a642324.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11123/HighFidelity-Beta-Interface-PR13543-a642324.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11123/HighFidelity-Beta-PR13543-a642324.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11123/HighFidelity-Beta-Interface-PR13543-a642324.exe

File: scripts/system/libraries/entitySelectionTool.js [hunk 1]
```diff
@@ -383,8 +383,6 @@ SelectionDisplay = (function() {
 
     var ctrlPressed = false;
 
-    var replaceCollisionsAfterStretch = false;
```
Comment (dback2) [inline scripts/system/libraries/entitySelectionTool.js]: This should not be removed

File: scripts/system/libraries/entitySelectionTool.js [hunk 2]
```diff
@@ -1265,8 +1260,8 @@ SelectionDisplay = (function() {
             var scaleRTFCubeToCamera = getDistanceToCamera(scaleRTFCubePosition);
             
             var scaleCubeToCamera = Math.min(scaleLBNCubeToCamera, scaleRBNCubeToCamera, scaleLBFCubeToCamera, 
-                                             scaleRBFCubeToCamera, scaleLTNCubeToCamera, scaleRTNCubeToCamera,
```
Comment (dback2) [inline scripts/system/libraries/entitySelectionTool.js]: I assume this is probably a change due to eslint complaining here, however I would say that the previous alignment is more in line with our coding standards https://docs.highfidelity.com/build-guide/coding-standards#2-1-6-the-incompleteness-of-split-lines-must-be-made-obvious and I have thus ignored eslint for cases like this here.

Same goes for the rest of the alignment changes below this.

File: scripts/system/libraries/entitySelectionTool.js (hunk 1 above)
Comment (ctrlaltdavid) [reply scripts/system/libraries/entitySelectionTool.js]: Actually, it should be:
```
that.replaceCollisionsAfterStretch = false;
```

File: scripts/system/libraries/entitySelectionTool.js (hunk 2 above)
Comment (ctrlaltdavid) [reply scripts/system/libraries/entitySelectionTool.js]: https://docs.highfidelity.com/build-guide/coding-standards#4-2-8-multi-line-statements-must-have-all-n-1-lines-indented-at accepts either indent scheme. C++ and JavaScript code both use both (and other schemes).

Personally, I prefer working in JavaScript with a minimum of extraneous ESLint errors. However, I'll revert to the original indentation.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11159/HighFidelity-Beta-PR13543-1a18d33.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11159/HighFidelity-Beta-PR13543-1a18d33.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11159/HighFidelity-Beta-Interface-PR13543-1a18d33.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11159/HighFidelity-Beta-PR13543-1a18d33.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11159/HighFidelity-Beta-Interface-PR13543-1a18d33.exe

**Comment by QliCknox** [comment]
currently testing

**Comment by QliCknox** [comment]
Pass @conklin94122 

Snapping to grid is working as intended

**Pull request merged**